src/solvers/dgsem_tree/subcell_limiters.jl

```diff
@@ -52,9 +52,15 @@ function SubcellLimiterIDP(equations::AbstractEquations, basis;
                            positivity_variables_cons = [],
                            positivity_correction_factor = 0.1)
     positivity = (length(positivity_variables_cons) > 0)
-    number_bounds = length(positivity_variables_cons)
 
-    cache = create_cache(SubcellLimiterIDP, equations, basis, number_bounds)
+    bounds_order = Dict()
```

Comment:
I think if you want to truly speed things up, you should consider restricting the dict type to something like `Dict{Symbol,Int}`. Right now, a lot of speed is probably lost because it is a `Dict{Any,Any}`.